service/src/main/java/bio/terra/pipelines/app/controller/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,92 @@
+package bio.terra.pipelines.app.controller;
+
+import bio.terra.common.exception.ErrorReportException;
+import bio.terra.pipelines.generated.model.ApiErrorReport;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.retry.backoff.BackOffInterruptedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+// This module provides a top-level exception handler for controllers.
+// All exceptions that rise through the controllers are caught in this handler.
+// It converts the exceptions into standard ApiErrorReport responses.
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+  private final Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
+
+  // -- Error Report - one of our exceptions --
+  @ExceptionHandler(ErrorReportException.class)
+  public ResponseEntity<ApiErrorReport> errorReportHandler(ErrorReportException ex) {
+    return buildApiErrorReport(ex, ex.getStatusCode(), ex.getCauses());
+  }
+
+  // -- validation exceptions - we don't control the exception raised
+  @ExceptionHandler({
+    MethodArgumentNotValidException.class,
+    MethodArgumentTypeMismatchException.class,
+    HttpMessageNotReadableException.class,
+    HttpRequestMethodNotSupportedException.class,
+    IllegalArgumentException.class,
+    NoHandlerFoundException.class
+  })
+  public ResponseEntity<ApiErrorReport> validationExceptionHandler(Exception ex) {
+    logger.error("Global exception handler: catch stack", ex);
+    // For security reasons, we generally don't want to include the user's invalid (and potentially
+    // malicious) input in the error response, which also means we don't include the full exception.
+    // Instead, we return a generic error message about input validation.
+    String validationErrorMessage =
+        "Request could not be parsed or was invalid: "
+            + ex.getClass().getSimpleName()
+            + ". Ensure that all types are correct and that enums have valid values.";
+    ApiErrorReport errorReport =
+        new ApiErrorReport()
+            .message(validationErrorMessage)
+            .statusCode(HttpStatus.BAD_REQUEST.value());
+    return new ResponseEntity<>(errorReport, HttpStatus.BAD_REQUEST);
+  }
+
+  // Exception thrown by Spring Retry code when interrupted.
+  @ExceptionHandler({BackOffInterruptedException.class})
```

Comment:
note we're not actively using Spring Retry yet - we don't use @Retryable yet - can note this for now, we will likely use it later